src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,120 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int INCREASE_CAPACITY = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == table.length * LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> current = table[getIndex(key)];
+        if (current == null) {
+            table[getIndex(key)] = newNode;
```

Comment:
it is better to move getIndex(key) into a local variable so we don't call this method 2 times